core/src/test/scala/org/apache/spark/SparkContextSuite.scala

```diff
@@ -1423,6 +1423,41 @@ class SparkContextSuite extends SparkFunSuite with LocalSparkContext with Eventu
     sc = new SparkContext(conf)
     sc.stop()
   }
+
+  test("SPARK-50247: BLOCK_MANAGER_REREGISTRATION_FAILED should be counted as network failure") {
```

Comment:
For the record, this test case has the same structure with the existing `HEARTBEAT_FAILURE` test case.

Reply:
Would you like to add such comment in the test? Without referring to the test, this test is hard to understand.

Reply:
Sure, @viirya